lib/iris/cube.py

```diff
@@ -3175,8 +3176,41 @@ def create_coords(src_coords, add_coord):
                     add_coord(result_coord, dims)
                     coord_mapping[id(src_coord)] = result_coord
 
+            def create_metadata(src_metadatas, add_metadata, metadata_type):
+                if metadata_type == "cell_measure":
+                    metadata_search_object = cube.Cube.cell_measure
+                elif metadata_type == "ancillary_var":
+                    metadata_search_object = cube.Cube.ancillary_variable
+                else:
+                    raise ValueError
```

Comment:
The ValueError probably ought to have some content.  

But then, **I'm thinking we don't really need it** , because ...
I think it would be simpler to just make the "metadata_search_object" another method argument,
instead of using an identifying string.
So ```def create_metadata(src_metadatas, add_metadata, metadata_search_object):```
and then change the calls (see below)

Given that, I think it would _also_ be clearer to rename "metadata_search_object" as "get_metadata"
(or something like), since that matches "add_metadata" : they are both method arguments.